### ISSUE TITLE
Can't download a file

### ISSUE THREAD

**Issue by hamsterrrr** [kevineye/dvrip-ctl#1]
I get this
```
laptop:~/Downloads/dvrip-ctl$ ./dvrip-ctl.pl -user admin -pass 'pass' -host 192.168.100.9 -port 34567  ls '2020-10-21 00:05:00' '2020-10-21 00:20:00' ;echo

[2020-10-22 05:33:31.87366] [77768] [info] connected to 192.168.100.9:34567
[2020-10-22 05:33:31.87395] [77768] [debug] ==> sending 107 byte login_req request
[2020-10-22 05:33:31.87699] [77768] [debug] <== received complete header: {"Channel":0,"Content_Length":128,"EndFlag":0,"MessageId":"login_rsp","Message_Code":1001,"Sequence":0,"SessionID":11,"Version":1}
[2020-10-22 05:33:31.87718] [77768] [debug] <== received login_rsp packet with 128 bytes of data
[2020-10-22 05:33:31.87743] [77768] [debug] ==> sending 193 byte filesearch_req request
[2020-10-22 05:33:31.88083] [77768] [debug] <== received complete header: {"Channel":0,"Content_Length":290,"EndFlag":0,"MessageId":"filesearch_rsp","Message_Code":1441,"Sequence":1,"SessionID":11,"Version":1}
[2020-10-22 05:33:31.88103] [77768] [debug] <== received filesearch_rsp packet with 290 bytes of data
[{"BeginTime":"2020-10-21 00:10:20","DiskNo":0,"EndTime":"2020-10-21 00:10:29","FileLength":"0x000006E3","FileName":"\/idea0\/2020-10-21\/001\/00.10.20-00.10.29[M][@71a][0].h264","SerialNo":0}]

laptop:~/Downloads/dvrip-ctl$ ./dvrip-ctl.pl -user admin -pass 'pass' -host 192.168.100.9 -port 34567 download {"BeginTime":"2020-10-21 00:10:20","DiskNo":0,"EndTime":"2020-10-21 00:10:29","FileLength":"0x000006E3","FileName":"\/idea0\/2020-10-21\/001\/00.10.20-00.10.29[M][@71a][0].h264","SerialNo":0}
[2020-10-22 05:33:53.22909] [77769] [info] connected to 192.168.100.9:34567
[2020-10-22 05:33:53.22955] [77769] [debug] ==> sending 107 byte login_req request
[2020-10-22 05:33:53.23251] [77769] [debug] <== received complete header: {"Channel":0,"Content_Length":128,"EndFlag":0,"MessageId":"login_rsp","Message_Code":1001,"Sequence":0,"SessionID":12,"Version":1}
[2020-10-22 05:33:53.23271] [77769] [debug] <== received login_rsp packet with 128 bytes of data
Can't use string ("BeginTime:2020-10-21 00:10:20") as a HASH ref while "strict refs" in use at /home/r/Downloads/dvrip-ctl/lib/IPCam.pm line 964.```

**Comment by kevineye** [comment]
1. Make sure the JSON argument is quoted for the shell.
2. Try removing the escaping that ends up in "FileName" -- the backslashes. I'm not actually sure if the camera sends it like that or if that's part of the JSON being escaped on output for display.
3. Only the BeginTime, EndTime and FileName parts of the JSON argument are used by the script so you may as well remove the others (though they shouldn't get in the way).

So basically, does this work?

    ./dvrip-ctl.pl -user admin -pass 'pass' -host 192.168.100.9 -port 34567 download '{"BeginTime":"2020-10-21 00:10:20","EndTime":"2020-10-21 00:10:29","FileName":"/idea0/2020-10-21/001/00.10.20-00.10.29[M][@71a][0].h264"}'

If not, maybe your camera is just a bit different than mine. I've only gotten to try this with exactly one model of camera. 🤷‍♂️

**Comment by hamsterrrr** [comment]
Yes, thank you. This did the trick. It is working now. Could you somehow change the software to filter out anything that doesn't fit 'BeginTime, EndTime and FileName parts of the JSON argument'?

**Comment by kevineye** [comment]
It basically does filter out those extra args, or doesn't use them for anything. I suggested removing them just because they make the JSON longer and introduce more opportunities for it to be malformed or not escaped correctly.
